src/viking.js

```diff
@@ -1,11 +1,98 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    }
+    return `${this.name} has died in act of combat`;
+  }
+  battleCry() {
+    return `Odin Owns You All!`;
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+  }
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    }
+    return `A Saxon has died in combat`;
+  }
+}
 
 // War
-class War {}
+class War {
+  vikingArmy = [];
+  saxonArmy = [];
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+  vikingAttack() {
+    // choose random sax
+    let vikingAttacker = this.vikingArmy[
+      Math.floor(Math.random() * this.vikingArmy.length)
+    ];
+    let saxonDefender = this.saxonArmy[
+      Math.floor(Math.random() * this.saxonArmy.length)
+    ];
+    let damage = saxonDefender.receiveDamage(vikingAttacker.strength);
+    if (saxonDefender.health <= 0) {
+      let i = this.saxonArmy.indexOf(saxonDefender);
```

Comment:
There appears to be a bit of double work going on here. On line 64, the index is generated by using the Math methods. Could, perhaps, the value of this generation be stored in a variable? This variable could be used on line 64 and in the splice method.